gateway/service/src/test/java/org/eclipse/ditto/gateway/service/endpoints/routes/thingsearch/ThingSearchRouteTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.gateway.service.endpoints.routes.thingsearch;
+
+import akka.http.javadsl.model.ContentTypes;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.testkit.TestRoute;
+import org.eclipse.ditto.gateway.service.endpoints.EndpointTestBase;
+import org.eclipse.ditto.json.JsonKey;
+import org.eclipse.ditto.json.JsonObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.eclipse.ditto.json.assertions.DittoJsonAssertions.assertThat;
+
+/**
+ * Builder for creating Akka HTTP routes for {@code /search/things}.
+ */
+public final class ThingSearchRouteTest extends EndpointTestBase {
+
+    private ThingSearchRoute thingsSearchRoute;
+    private TestRoute underTest;
+
+
+    @Before
+    public void setUp() {
+        thingsSearchRoute = new ThingSearchRoute(routeBaseProperties);
+        final Route route = extractRequestContext(ctx -> thingsSearchRoute.buildSearchRoute(ctx, dittoHeaders));
+        underTest = testRoute(handleExceptions(() -> route));
+    }
+
+    @Test
+    public void searchThingsShouldGetFilterFromBody() {
+        final var filter = "and(like(definition,\"*test*\"))";
+        final var result = underTest.run(HttpRequest.POST("/search/things")
+                .withEntity(ContentTypes.APPLICATION_JSON, filter));
+        result.assertStatusCode(StatusCodes.OK);
+        assertThat(JsonObject.of(result.entityString()))
+                .contains(
+                        JsonKey.of("filter"),
+                        filter
+                );
+    }
+
+    @Test
+    public void countThingsShouldGetFilterFromBody() {
+        final var filter = "and(like(definition,\"*test*\"))";
+        final var result = underTest.run(HttpRequest.POST("/search/things/count")
+                .withEntity(ContentTypes.APPLICATION_JSON, filter));
```

Comment:
Please also adjust to `application/x-www-form-urlencoded` content-type.

Reply:
done